### PR TITLE
bug fix #2135

### LINKED ISSUE #2135: bug: client crashes when hovering over an item in the hotkey
### Description

client crashes when hovering over an item in the hotkey, when the item is not in your backpack either.

### Steps to Reproduce

Have the item in your inventory.
Drag to the hotbar.
Store the item in the bank.
Hover over the item in the hotbar.
see the bug

### Version with bug

0.8.0.0

### Last version that worked well

Unknown

### Affected platforms

Windows, I was *not* able test on other platforms

### Did you find any workaround?

_No response_

### Relevant log output

```shell
2023-12-03 23:29:14.557 [Error] NullReferenceException: Object reference not set to an instance of an object.
    Stack:    at Intersect.Client.Interface.Game.DescriptionWindows.ItemDescriptionWindow.SetupEquipmentInfo() in E:\The Celtos Game\Engine\Intersect-Engine\Intersect.Client\Interface\Game\DescriptionWindows\ItemDescriptionWindow.cs:line 209
   at Intersect.Client.Interface.Game.DescriptionWindows.ItemDescriptionWindow.SetupDescriptionWindow() in E:\The Celtos Game\Engine\Intersect-Engine\Intersect.Client\Interface\Game\DescriptionWindows\ItemDescriptionWindow.cs:line 99
   at Intersect.Client.Interface.Game.DescriptionWindows.ItemDescriptionWindow..ctor(ItemBase item, Int32 amount, Int32 x, Int32 y, ItemProperties itemProperties, String titleOverride, String valueLabel, Boolean centerOnPosition) in E:\The Celtos Game\Engine\Intersect-Engine\Intersect.Client\Interface\Game\DescriptionWindows\ItemDescriptionWindow.cs:line 54
   at Intersect.Client.Interface.Game.Hotbar.HotbarItem.pnl_HoverEnter(Base sender, EventArgs arguments) in E:\The Celtos Game\Engine\Intersect-Engine\Intersect.Client\Interface\Game\Hotbar\HotbarItem.cs:line 183
   at Intersect.Client.Framework.Gwen.Control.Base.OnMouseEntered() in E:\The Celtos Game\Engine\Intersect-Engine\Intersect.Client.Framework\Gwen\Control\Base.cs:line 2240
   at Intersect.Client.Framework.Gwen.Input.InputHandler.UpdateHoveredControl(Base inCanvas) in E:\The Celtos Game\Engine\Intersect-Engine\Intersect.Client.Framework\Gwen\Input\InputHandler.cs:line 542
   at Intersect.Client.Framework.Gwen.Control.Canvas.Input_MouseMoved(Int32 x, Int32 y, Int32 dx, Int32 dy) in E:\The Celtos Game\Engine\Intersect-Engine\Intersect.Client.Framework\Gwen\Control\Canvas.cs:line 310
   at Intersect.Client.Framework.Gwen.Input.IntersectInput.ProcessMessage(Object message) in E:\The Celtos Game\Engine\Intersect-Engine\Intersect.Client.Framework\Gwen\Input\IntersectInput.cs:line 96
   at Intersect.Client.MonoGame.Input.MonoInput.Update(TimeSpan elapsed) in E:\The Celtos Game\Engine\Intersect-Engine\Intersect.Client\MonoGame\Input\MonoInput.cs:line 405
   at Intersect.Client.Core.Main.Update(TimeSpan deltaTime) in E:\The Celtos Game\Engine\Intersect-Engine\Intersect.Client\Core\Main.cs:line 111
   at Intersect.Client.MonoGame.IntersectGame.Update(GameTime gameTime) in E:\The Celtos Game\Engine\Intersect-Engine\Intersect.Client\MonoGame\IntersectGame.cs:line 284
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime)
   at Microsoft.Xna.Framework.Game.Tick()
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop()
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior)
   at Microsoft.Xna.Framework.Game.Run()
   at Intersect.Client.MonoGame.IntersectGame.MonoGameRunner.Start(IClientContext context, Action postStartupAction) in E:\The Celtos Game\Engine\Intersect-Engine\Intersect.Client\MonoGame\IntersectGame.cs:line 566
   at Intersect.Client.Core.ClientContext.InternalStart() in E:\The Celtos Game\Engine\Intersect-Engine\Intersect.Client\Core\ClientContext.cs:line 41
   at Intersect.Core.ApplicationContext`2.Start(Boolean lockUntilShutdown) in E:\The Celtos Game\Engine\Intersect-Engine\Intersect (Core)\Core\ApplicationContext`2.cs:line 239


--------------------------------------------------------------------------------
```

### Duplicate Bug Check

- [X] This bug report is not a duplicate to the best of my knowledge.

### PR DESCRIPTION
**Pull request by TimtyG**
Fixes #2135 
 
I've added a null check for the range.

This change ensures that the code handles scenarios where the range object is null and prevents unexpected client crashes.

### CODE REVIEW

File: Intersect.Client/Interface/Game/DescriptionWindows/ItemDescriptionWindow.cs [hunk 1]
```diff
@@ -354,19 +354,21 @@ protected void SetupEquipmentInfo()
                 else
                 {
                     _ = mItem.TryGetRangeFor((Stat)i, out var range);
```
Comment (lodicolo) [inline Intersect.Client/Interface/Game/DescriptionWindows/ItemDescriptionWindow.cs:356]: You don't need a separate null check, the boolean being discarded here will already tell you that

File: Intersect.Client/Interface/Game/DescriptionWindows/ItemDescriptionWindow.cs [hunk 2]
```diff
@@ -354,19 +354,21 @@ protected void SetupEquipmentInfo()
...
-                    if (mItem.PercentageStatsGiven[i] != 0)
-                    {
-                        rows.AddKeyValueRow(Strings.ItemDescription.StatCounts[i], Strings.ItemDescription.RegularAndPercentage.ToString(Strings.ItemDescription.StatGrowthRange.ToString(statLow, statHigh), mItem.PercentageStatsGiven[i]));
-                    }
-                    else
+                    if (range != null)
                     {
-                        rows.AddKeyValueRow(Strings.ItemDescription.StatCounts[i], Strings.ItemDescription.StatGrowthRange.ToString(statLow, statHigh));
+                        var statLow = mItem.StatsGiven[i] + range.LowRange;
+                        var statHigh = mItem.StatsGiven[i] + range.HighRange;
```
Comment (lodicolo) [inline Intersect.Client/Interface/Game/DescriptionWindows/ItemDescriptionWindow.cs:360]: Cache `mItem.StatsGivern[i]` in a local variable

File: Intersect.Client/Interface/Game/DescriptionWindows/ItemDescriptionWindow.cs [hunk 3]
```diff
@@ -354,19 +354,21 @@ protected void SetupEquipmentInfo()
...
+
+                        if (mItem.PercentageStatsGiven[i] != 0)
+                        {
+                            rows.AddKeyValueRow(Strings.ItemDescription.StatCounts[i], Strings.ItemDescription.RegularAndPercentage.ToString(Strings.ItemDescription.StatGrowthRange.ToString(statLow, statHigh), mItem.PercentageStatsGiven[i]));
```
Comment (lodicolo) [inline Intersect.Client/Interface/Game/DescriptionWindows/ItemDescriptionWindow.cs:364]: Cache `mItem.PercentageStatsGiven[i]` in another local variable

File: Intersect.Client/Interface/Game/DescriptionWindows/ItemDescriptionWindow.cs [hunk 4]
```diff
@@ -354,19 +354,21 @@ protected void SetupEquipmentInfo()
...
+                        }
+                        else
+                        {
+                            rows.AddKeyValueRow(Strings.ItemDescription.StatCounts[i], Strings.ItemDescription.StatGrowthRange.ToString(statLow, statHigh));
```
Comment (lodicolo) [inline Intersect.Client/Interface/Game/DescriptionWindows/ItemDescriptionWindow.cs:368]: Use the if statement to set the message, only call `rows.AddKeyValueRow(Strings.ItemDescription.StatCounts[i], ...)` once

**Review by lodicolo — CHANGES_REQUESTED** [review]

File: Intersect.Client/Interface/Game/DescriptionWindows/ItemDescriptionWindow.cs (hunk 2 above)
Comment (TimtyG) [reply Intersect.Client/Interface/Game/DescriptionWindows/ItemDescriptionWindow.cs]: Done

File: Intersect.Client/Interface/Game/DescriptionWindows/ItemDescriptionWindow.cs [hunk 5]
```diff
@@ -353,20 +353,27 @@ protected void SetupEquipmentInfo()
...
+
+                        string statMessage;
+
+                        if (percentageStatGiven != 0)
+                        {
+                            statMessage = Strings.ItemDescription.RegularAndPercentage.ToString(Strings.ItemDescription.StatGrowthRange.ToString(statLow, statHigh), percentageStatGiven);
+                        }
+                        else
+                        {
+                            statMessage = Strings.ItemDescription.StatGrowthRange.ToString(statLow, statHigh);
```
Comment (lodicolo) [inline Intersect.Client/Interface/Game/DescriptionWindows/ItemDescriptionWindow.cs:371]: Correct me if I'm wrong, but I think I'm seeing `Strings.ItemDescription.StatGrowthRange.ToString(statLow, statHigh)` computed in both arms of the if statement?

So you can instead do `var statMessage = Strings.ItemDescription.StatGrowthRange.ToString(statLow, statHigh);` before the if statement, and do `statMessage = ....ToString(statMessage, percentageStatGiven)` in the `if` arm (delete the `else` arm).

**Review by lodicolo — CHANGES_REQUESTED** [review]

File: Intersect.Client/Interface/Game/DescriptionWindows/ItemDescriptionWindow.cs (hunk 5 above)
Comment (TimtyG) [reply Intersect.Client/Interface/Game/DescriptionWindows/ItemDescriptionWindow.cs]: You're right, thanks

**Review by lodicolo — APPROVED** [review]

**Pull request merged**